File: c++/263. Ugly Number.cpp
class Solution {
public:
    bool isUgly(int num) {
        int &a = num;
        if (a <= 0) return false;
        while (a % 2 == 0) a /= 2;
        while (a % 3 == 0) a /= 3;
        while (a % 5 == 0) a /= 5;
        return (a == 1);
    }
};	

File: go/src/leetcode/263. Ugly Number/263. Ugly Number.go
package _63__Ugly_Number

func isUgly(n int) bool {
	for n%2 == 0 {
		n /= 2
	}
	for n%3 == 0 {
		n /= 3
	}
	for n %5 ==0 {
		n /= 5
	}
	return n==1
}